### ISSUE TITLE
Different feature folder

### ISSUE THREAD

**Issue by AFASbart** [jonsamwell/flutter_gherkin#174]
When setting the features or the feature path (or both), it still imports all features under the features folder. I want the structure features/casex and features/casey. When I enter the path features/casey it still includes the features of the casex with it.

**Comment by jonsamwell** [comment]
I'm not entirely sure what you mean here.  Can you give a worked example?

**Comment by AFASbart** [comment]
```
Future<void> main() async {
  executeTestSuite(
    FlutterTestConfiguration.DEFAULT(StepManager().getSteps(),
        featurePath: "integration_test/features/DM_ERP/*.*.feature")
      ..reporters = [
        StdoutReporter(MessageLevel.error)
          ..setWriteLineFn(print)
          ..setWriteFn(print),
        ProgressReporter()
          ..setWriteLineFn(print)
          ..setWriteFn(print),
        TestRunSummaryReporter()
          ..setWriteLineFn(print)
          ..setWriteFn(print),
        JsonReporter(
          writeReport: (_, __) => Future<void>.value(),
        ),
      ]
      ..hooks = [
        AttachScreenshotOnFailedStepHook(),
        PinHook(),
        SignInHook(),
        RestoreHook(),
        ErrorHook(),
      ]
      ..featureDefaultLanguage = "en"
      ..defaultTimeout = const Duration(seconds: 30)
      ..order = ExecutionOrder.sequential
      ..features = [RegExp(r'features\\DM_ERP\\.*\.feature')],
    (World world) async {
      WidgetsFlutterBinding.ensureInitialized();

      runApp(
        InitializingApp(
            key: UniqueKey(),
            onInitializationComplete: () =>
                runApp(RestartWidget(child: PocketApp()))),
      );
    },
  );
}
```

So the path I am trying is: integration_test/features/DM_ERP/*.*.feature.
But it still grabs features from the integration_test/features/FLEX_POC/*.*.feature folder.

**Comment by jonsamwell** [comment]
Odd as it just assesses the pattern against the file path.  Are you able to debug to see what is going on?

https://github.com/jonsamwell/dart_gherkin/blob/master/lib/src/io/io_feature_file_accessor.dart#L46

I've never really like that code so it could do with some TLC

**Comment by AFASbart** [comment]
I am using the flutter_gherkin not the dark_gherkin? So where do I find it in the code of this package?

**Comment by jonsamwell** [comment]
flutter_gherkin sits on top of dart_gherkin.  It you set debugging to my code + packages and find that file above you should be able to debug.

I wonder if there is some conflict between featurePath & features in the above config

**Comment by AFASbart** [comment]
For some weird reason my debug is not working, I am using the flutter command line tool then it will run the tests. Is it possible for me to access the file so I can add an print() there?

**Comment by jonsamwell** [comment]
I think you could modify the local version (although I wouldn't recommend it!).  Is the debugger set to debug packages as well?

VSCode shows this:

![image](https://user-images.githubusercontent.com/3694497/142396499-c74e3f5c-8607-4b13-9631-382976c57133.png)

**Comment by AFASbart** [comment]
I have that enabled, but I cannot use the debugger because for one reason it won't go into an hook that enters a pin code.
It just start-up the app and then does nothing.

**Comment by AFASbart** [comment]
I know why I can't use the debugger. I have an websocket server for communication to my localhost pc. The websocket can't connect and times out...

**Comment by AFASbart** [comment]
What launch config are you using? With me it doesn't enter the integration_test_driver.dart main() function.

**Comment by AFASbart** [comment]
@jonsamwell Do you have the launch.json file that you are using for me? In my project it doesn't use the integration_Test_driver.dart file and I need it because I am setting up some important things like the websocket and it saves the report.

**Comment by jonsamwell** [comment]
@AFASbart this is a launch.json that I use in on of my project 

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Debug (Dev)",
            "request": "launch",
            "type": "dart",
            "program": "lib/main_dev.dart",
            "args": [
                "--flavor",
                "development"
            ],
            "flutterMode": "debug"
        },
        {
            "name": "Debug (Local)",
            "request": "launch",
            "type": "dart",
            "program": "lib/main_local.dart",
            "args": [
                "--flavor",
                "development"
            ],
            "flutterMode": "debug"
        },
        {
            "name": "Debug integration tests (dev)",
            "program": "test/integration/test_driver.dart",
            "request": "launch",
            "type": "dart",
            "args": [
                "--target=test/integration/suites/gherkin_suite_test.dart",
                "--flavor=development",
                "--dart-define=targetConfig=dev",
                "--dart-define=tagExpression=@debug",
                "--dart-define=disableAppNotifications=false",
            ],
        },
        {
            "name": "Debug integration tests (local)",
            "program": "test/integration/test_driver.dart",
            "request": "launch",
            "type": "dart",
            "args": [
                "--target=test/integration/suites/gherkin_suite_test.dart",
                "--flavor=development",
                "--dart-define=targetConfig=local",
                "--dart-define=tagExpression=@debug",
                "--dart-define=disableAppNotifications=false",
            ],
        }
    ]
}
```

This my gherkin_suite_test.dart file

```
import 'package:app/dependency_config.dart';
import 'package:flutter_gherkin/flutter_gherkin.dart';
import 'package:gherkin/gherkin.dart';

import '../test_configuration.dart';

part 'gherkin_suite_test.g.dart';

@GherkinTestSuite(featurePaths: <String>[
  'test/integration/features/**.feature',
])
void main() {
  const targetConfig = String.fromEnvironment(
    'targetConfig',
    defaultValue: 'dev',
  );
  const tagExpression = String.fromEnvironment(
    'tagExpression',
    defaultValue: 'not @ignore',
  );
  const disableAppNotifications =
      bool.fromEnvironment('disableAppNotifications');
  const screenshotDirectory = String.fromEnvironment(
    'screenshotDirectory',
    defaultValue: './screenshots',
  );
  const deviceName = String.fromEnvironment(
    'deviceName',
    defaultValue: 'emulator',
  );

  // ignore: avoid_print
  print('*************** TEST CONFIGURATION ********************');
  // ignore: avoid_print
  print('Config: `$targetConfig`');
  // ignore: avoid_print
  print('Tag expression: `$tagExpression`');
  // ignore: avoid_print
  print('Notifications disabled: `$disableAppNotifications`');
  // ignore: avoid_print
  print('*******************************************************');

  executeTestSuite(
    getConfiguration(
      targetConfig,
      tagExpression,
      disableAppNotifications,
      deviceName,
      screenshotDirectory,
    ),
    appInitializationFn(
      targetConfig == 'dev'
          ? ContainerConfiguration.dev
          : ContainerConfiguration.local,
    ),
  );
}

```

**Comment by AFASbart** [comment]
Ok, so the different feature folder is kinda fixed? If I give the feature path with the @GherkinTestSuite then it works. But in the conf itself it doesn't work on generation of the features.
For the integration_test_driver and launch.json part, I am going to work on that now.

**Comment by jonsamwell** [comment]
@AFASbart I am not entirely sure what you mean?

**Comment by AFASbart** [comment]
This is my configuration (gherkin_test_suite.dart)
```
import 'dart:io';

import 'package:afas_pocket/main.dart';
import 'package:afas_pocket/pocketApp.dart';
import 'package:flutter/widgets.dart';
import 'package:flutter_gherkin/flutter_gherkin.dart';
import 'package:gherkin/gherkin.dart';

import 'hooks/errorHook.dart';
import 'hooks/pinHook.dart';
import 'hooks/signInHook.dart';
import 'hooks/snapshotHook.dart';
import 'steps/stepManager.dart';

part 'gherkin_suite_test.g.dart';

@GherkinTestSuite(featurePaths: <String>[
  'integration_test/features/DM_ERP/**.feature',
])
Future<void> main() async {
  executeTestSuite(
    FlutterTestConfiguration.DEFAULT(StepManager().getSteps())
      ..reporters = [
        StdoutReporter(MessageLevel.error)
          ..setWriteLineFn(print)
          ..setWriteFn(print),
        ProgressReporter()
          ..setWriteLineFn(print)
          ..setWriteFn(print),
        TestRunSummaryReporter()
          ..setWriteLineFn(print)
          ..setWriteFn(print),
        JsonReporter(
          writeReport: (_, __) => Future<void>.value(),
        ),
      ]
      ..hooks = [
        AttachScreenshotOnFailedStepHook(),
        PinHook(),
        SignInHook(),
        RestoreHook(),
        ErrorHook(),
      ]
      // ..featureDefaultLanguage = "en"
      ..defaultTimeout = const Duration(seconds: 30)
      ..order = ExecutionOrder.sequential,
    (World world) async {
      WidgetsFlutterBinding.ensureInitialized();

      runApp(
        InitializingApp(
            key: UniqueKey(),
            onInitializationComplete: () =>
                runApp(RestartWidget(child: PocketApp()))),
      );
    },
  );
}
```

And it had to be this ```
@GherkinTestSuite(featurePaths: <String>[
  'integration_test/features/DM_ERP/**.feature',
])```

Instead of an parameter (like ..defaultTimeout) in ```FlutterTestConfiguration.DEFAULT```

**Comment by AFASbart** [comment]
I have fixed this issue with the pull-request #187
